src/services/walletConnect.js

```diff
@@ -39,9 +56,42 @@ const clientMeta = {
 };
 /* eslint-enable i18next/no-literal-string */
 
-export const createConnector = (
-  options: WalletConnectOptions,
-): ?WalletConnectConnector => {
+export async function createWeb3Wallet() {
+  core = new Core({
+    projectId: getEnv().PROJECT_ID,
+  });
+
+  web3wallet = await Web3Wallet.init({
+    core,
+    metadata: clientMeta,
+  });
+
+  return web3wallet;
+}
+
+export async function web3WalletInit() {
+  const newCore = new Core({
+    projectId: getEnv().PROJECT_ID,
+  });
+
+  const wcInit = await Web3Wallet.init({
```

Comment:
We have two `Web3Wallet.init` calls here - can we not reuse this in some way? I'm just worried that we end up with two instances of Web3Wallet which may cause connection issues

Reply:
Please let me know if there was a reason for the above

Reply:
The session disconnect method in need new instance. So I made it separate. first one (createWeb3Wallet) used for every methods.

Reply:
Understood, thanks!